ngx_rtmp_live_module.c

```diff
@@ -504,6 +504,19 @@ ngx_rtmp_live_join(ngx_rtmp_session_t *s, u_char *name, unsigned publisher)
 
     if (ctx == NULL) {
         ctx = ngx_palloc(s->connection->pool, sizeof(ngx_rtmp_live_ctx_t));
+        
+        if (ctx == NULL) {
+            ngx_log_error(NGX_LOG_ERR, s->connection->log, 0,
+                          "live: failed to allocate memory for context");
+
+            ngx_rtmp_send_status(s, "NetStream.Play.Failed", "error",
```

Comment:
Not only `Play`, `Publish` also calls this function, so it should be:

```
            if (publisher) {
                ngx_rtmp_send_status(s, "NetStream.Publish.Failed", "error",
                                     "Failed to allocate memory");
            } else {
                ngx_rtmp_send_status(s, "NetStream.Play.Failed", "error",
                                     "Failed to allocate memory");
            }
```

I will merge into my own fork :)